src/main/java/com/example/bookstore/middlewares/SecurityMiddleware.java

```diff
@@ -0,0 +1,49 @@
+package com.example.bookstore.middlewares;
+
+import com.example.bookstore.facades.UserFacade;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.security.authentication.dao.DaoAuthenticationProvider;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+
+@EnableWebSecurity
+public class SecurityMiddleware extends WebSecurityConfigurerAdapter {
+
+    @Autowired
+    private UserFacade userFacade;
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http
+                .authorizeRequests()
+                .anyRequest().permitAll()
+                .and().httpBasic()
+                .and().csrf().disable();
+    }
+
+    @Override
+    protected void configure(AuthenticationManagerBuilder auth)
+            throws Exception {
+        auth.authenticationProvider(authenticationProvider());
+    }
+
+    @Bean
+    public DaoAuthenticationProvider authenticationProvider() {
+        DaoAuthenticationProvider authProvider
+                = new DaoAuthenticationProvider();
+        authProvider.setUserDetailsService(userFacade);
+        authProvider.setPasswordEncoder(encoder());
+        return authProvider;
+    }
+
+    @Bean
+    public PasswordEncoder encoder() {
+        return new BCryptPasswordEncoder(11);
```

Comment:
petit commentaire pourquoi 11 ? Voir tu peux l'extraire dans une private static final avec le nom qui va bien au niveau du service

Reply:
Ah j'ai pris cette encoder sur un site qui mettait 11 par défault. Mais en recherchant un peu. >12 est mieux. 
https://security.stackexchange.com/questions/17207/recommended-of-rounds-for-bcrypt

Reply:
J'ai bien lu la doc, regarde du côté de "magic number", ma remarque porte surtout là dessus..